.github/workflows/maven_test.yml

```diff
@@ -178,13 +184,7 @@ jobs:
       # Run the tests.
       - name: Run tests
         env: ${{ fromJSON(inputs.envs) }}
-        # The command script takes different options ubuntu vs macos-14, see also SPARK-47095.
-        shell: '[[ "${{ inputs.os }}" == *"ubuntu"* ]] && script -q -e -c "bash {0}" || script -q -e "bash {0}"'
```

Comment:
Few notes:
- The main problem here is that `inputs.os` isn't available within `shell` (see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability). 
  - I can't work around via environment variable either that's not replaced properly
  - I can't create a custom command either because the path here should be an absolute path (home path also does not work).
- `script -q -e -c` is only workaround for ubuntu. There's no workaround for Mac (so `AmmoniteTest` fails because of TTY issue).